### PR TITLE
Create filtering for available products

### PR DESCRIPTION
**Pull request by Ashraf-Alshashaa**
You can filter products now by following multiple integrated filters:
- search
- max price
- min price
---
NOTE:
If u don't provide any query u will end up with an response validation error, this error has been fixed in [create-get-all-available-products](https://github.com/Ashraf-Alshashaa/Trade-Hub/tree/create-get-all-available-products)

### CODE REVIEW

File: db/db_product.py [hunk 1]
```diff
@@ -131,13 +131,51 @@ def choose_buyer(db: Session, bid_id: int):
...
-def search(db: Session, search_str: str):
-    products = db.query(DbProduct).filter(
-        (DbProduct.buyer_id == None) &
-        (DbProduct.name.ilike(f"%{search_str}%") |
-        DbProduct.description.ilike(f"%{search_str}%"))
-    ).all()
+def filter_available_products(
+        db: Session, search_str: str = None, 
+        max_price: int = None,
+        min_price: int = None
```
Comment (Laaad) [inline db/db_product.py:137]: If `search_str`,`max_price` and `min_price` are optional as you mentioned in the description then you should define them by the `Optional` keyword here as well.

File: db/db_product.py [hunk 2]
```diff
@@ -131,13 +131,51 @@ def choose_buyer(db: Session, bid_id: int):
...
+    - **min_price**: Optional[int]
+        Filter products with a price greater than or equal to the specified value. This is an optional integer parameter.
+
+    Returns:
+    - **List[ProductDisplay]**
+        List of products matching the specified filters, serialized as `ProductDisplay` Pydantic models.
+
+    Raises:
+    - **HTTP_404_NOT_FOUND**
+        If no products match the filters.
```
Comment (Laaad) [inline db/db_product.py:158]: Here I would rather see nothing or an empty list instead of an Error.

**Review by Laaad — CHANGES_REQUESTED** [review]
User should be able to apply both mix and max and yet get the result.
The logic used in `filtered_available_products` needs reconsideration.

File: db/db_product.py (hunk 1 above)
Comment (Ashraf-Alshashaa) [reply db/db_product.py]: done

File: db/db_product.py (hunk 2 above)
Comment (Ashraf-Alshashaa) [reply db/db_product.py]: done

**Pull request merged**